File: app/graphql/resolvers/update_pomodoro.rb
class Resolvers::UpdatePomodoro < GraphQL::Function
  # arguments passed as "args"
  argument :id, !types.ID
  argument :title, types.String
  argument :reflection, types.String
  argument :end_time, types.String
  # return type from the mutation
  type Types::PomodoroType

  # the mutation method
  # _obj - is parent object, which in this case is nil
  # args - are the arguments passed
  # _ctx - is the GraphQL context
  def call(_obj, args, _ctx)
    pomodoro = Pomodoro.find args[:id]

    pomodoro.update_attributes(
      title: args[:title],
      reflection: args[:reflection],
      end_time: args[:end_time]
    )
    pomodoro
  end
end

File: app/javascript/src/graphql/queries/updatePomodoro.ts
import gql from 'graphql-tag';

export default gql`
  mutation updatePomodoro(
    $id: ID!,
    $title: String,
    $reflection: String,
    $end_time: String
    
  ) {
    updatePomodoro(
      id: $id
      title: $title
      reflection: $reflection
      end_time: $end_time
    ) {
      id
      title
      reflection
      end_time
      created_at
      updated_at
    }
  }
`

